popup/js/search/fuzzySearch.js

```diff
@@ -34,48 +34,44 @@ export async function fuzzySearch(searchMode, searchTerm) {
 function fuzzySearchWithScoring(searchTerm, data) {
   /** Search results */
   const results = []
+  const haystack = data.map((el) => {
```

Comment:
ideally this step gets pre-generated and cached?

Reply:
Thanks for looking into the PR :)

Yeah, I'm not sure if this can be cached with the current architecture. Bascially a search is stateless and every time a search is done, it will reinitialize. The haystack is only mapped once per search for each different dataset to look over (bookmarks, tabs etc. are separated datasets and searches).

Reply:
But now after you mentioned it, maybe I can change the architecture so this can get properly cached (only done once per data loading), also the uFuzzy instance creation.

Reply:
if the search is "live" (as you type) then it might be worth it. if it's debounced and only searches after you've stopped rtping for 100ms, then probably not.

i'm a bit of a premature optimizer, but it's served me well so far in my career ;)

Reply:
Yeah, I can relate :) I've now memoized it.